src/formatDate.js

```diff
@@ -7,8 +7,39 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // Розділяємо дату на частини за допомогою роздільника у форматі fromFormat
+  const separator = fromFormat.pop(); // Останній елемент - це роздільник
+  const dateParts = date.split(separator);
+
+  // Створюємо об'єкт, що містить частини дати відповідно до fromFormat
+  const dateObj = {};
+
+  fromFormat.forEach((part, i) => {
+    dateObj[part] = dateParts[i];
+  });
+
+  // Конвертація формату року з YY в YYYY або назад
+  if (dateObj.YYYY === undefined && dateObj.YY) {
+    const year = Number(dateObj.YY);
+    // Додаємо перевірку для року '00'
+
+    if (year === 0) {
+      dateObj.YYYY = '2000';
+    } else {
+      dateObj.YYYY =
+        year < 30 ? '20' + String(year).padStart(2, '0') : '19' + year;
```

Comment:
The condition `year === 0` is not necessary since the task specifies that the input will always be correctly formatted, which implies that year `00` should be treated as `2000` following the rule for `YY` to `YYYY` conversion. You can remove this condition and handle all cases with the subsequent logic.